packages/browser/src/login/oidc/redirectHandler/AuthCodeRedirectHandler.ts

```diff
@@ -61,30 +62,53 @@ export default class AuthCodeRedirectHandler implements IRedirectHandler {
       );
     }
     const url = new URL(redirectUrl, true);
-    const sessionId = url.query.state as string;
-    const [codeVerifier, redirectUri] = await Promise.all([
-      (await this.storageUtility.getForUser(sessionId, "codeVerifier", {
-        errorIfNull: true,
-      })) as string,
-      (await this.storageUtility.getForUser(sessionId, "redirectUri", {
+    const oauthState = url.query.state as string;
+
+    const storedSessionId = (await this.storageUtility.getForUser(
+      oauthState,
+      "sessionId",
+      {
         errorIfNull: true,
-      })) as string,
-    ]);
+      }
+    )) as string;
+
+    // PMCB55: TODO: I think we still need a catch handler around this...
+    const signinResponse = await new OidcClient({
+      // We are instantiating a new instance here, so the only value we need to
+      // explicitly provide is the response mode (default otherwise will look
+      // for a hash '#' fragment!).
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      response_mode: "query",
+    }).processSigninResponse(redirectUrl.toString());
+
+    // We need to decode the access_token JWT to extract out the full WebID.
+    const decoded = await this.joseUtility.decodeJWT(
```

Comment:
Does oidc-client not provide something to decode JWTs? And if not, couldn't we use a library focused on supporting it? It would be good if we would be able to get rid of `node-jose` for `-browser` when all this is done.

Reply:
Yeah - will be doing this as a separate PR (basically moving towards calling crypto-libraries directly (i.e. making JoseUtility just a very little wrapper)).